pkg/operator/cluster/controller.go

```diff
@@ -166,7 +166,17 @@ func (c *ClusterController) onAdd(obj interface{}) {
 }
 
 func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
-	logger.Infof("modifying a cluster not implemented")
+	oldCluster := oldObj.(*Cluster)
```

Comment:
should all these type casts also be making a deep copy as mentioned here? https://github.com/rook/rook/blob/master/pkg/operator/cluster/controller.go#L105

i know we're not modifying the objects right now, but if we need that deep copy down the line it may be good to add it now so that someone doesn't come along later and forget to do it then.

this comment applies to other CRD controllers in this PR.

Reply:
you're right, we should make a deep copy. However, I think we can wait for the k8s code generation feature that is just around the corner with PR #1153 that will take care of this for us.

Reply:
i'm afraid i don't know much about the code generation for CRDs and how that would impact the need to use deep copies.  would making a deep copy no longer be necessary? would it just be part of the generated code?  would we still need to add code to do deep copies but it would look totally different? thank you for the education :)

Reply:
the code generation will actually generate the deep copies for us. Searching #1153 for `deepcopy` will show how much of the generated code is for that purpose.